perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -113,26 +127,179 @@ sub bin_services_for_address {
             next if $now_dt > $to_dt;
         }
 
-        my $containers = $self->_containers($property);
+        my ($round) = split / /, $service->{RoundSchedule};
 
-        push @site_services_filtered, {
+        my $filtered_service = {
             id             => $service->{SiteServiceID},
             service_id     => $service->{ServiceItemName},
             service_name   => $containers->{ $service->{ServiceItemName} },
             round_schedule => $service->{RoundSchedule},
+            round          => $round,
             next           => {
                 date    => $service->{NextCollectionDate},
                 ordinal => ordinal( $next_dt->day ),
                 changed => 0,
             },
         };
+
+        # Get the last collection date from recent collections
+        my $last_dt = $property->{recent_collections}{ $service->{RoundSchedule} };
+
+        if ($last_dt) {
+            $filtered_service->{last} = {
+                date    => $last_dt,
+                ordinal => ordinal( $last_dt->day ),
+            };
+        }
+
+        $filtered_service->{report_open}
+            = $property->{missed_collection_reports}{ $filtered_service->{service_id} } ? 1 : 0;
+
+        $filtered_service->{report_allowed}
+            = $self->can_report_missed( $property, $filtered_service );
+
+        $filtered_service->{report_locked_out}
+            = $property->{round_exceptions}{ $filtered_service->{round} } ? 1 : 0;
+
+        push @site_services_filtered, $filtered_service;
     }
 
     @site_services_filtered = $self->service_sort(@site_services_filtered);
 
     return \@site_services_filtered;
 }
 
+# Returns hashref of 'ServiceItemName's (FO-140, GA-140, etc.) that have
+# open missed collection reports against them on the given property
+sub _missed_collection_reports {
+    my ( $self, $property ) = @_;
+
+    # If property has parent, use that instead
+    my $uprn
+        = $property->{parent_property}
+        ? $property->{parent_property}{uprn}
+        : $property->{uprn};
+
+    my $worksheets = $self->whitespace->GetSiteWorksheets($uprn);
+
+    my %missed_collection_reports;
+    for my $ws (@$worksheets) {
+        if (   $ws->{WorksheetStatusName} eq 'Open'
+            && $ws->{WorksheetSubject} =~ /^Missed/ )
+        {
+            for ( @{  $self->whitespace->GetWorksheetDetailServiceItems(
```

Comment:
Can this be called with more than one ID? Guessing not, but would be fewer calls if it can.